pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -57,6 +57,11 @@ func GetCloudInfo(ic *types.InstallConfig) (*CloudInfo, error) {
 
 	opts := &clientconfig.ClientOpts{Cloud: ic.OpenStack.Cloud}
 
+	// We explicitly disable reading auth data from env variables by setting an invalid EnvPrefix.
+	// By doing this, we make sure that the data from clouds.yaml is enough to authenticate.
+	// For more information: https://github.com/gophercloud/utils/blob/8677e053dcf1f05d0fa0a616094aace04690eb94/openstack/clientconfig/requests.go#L508
+	opts.EnvPrefix = "NO_ENV_VARIABLES_"
```

Comment:
This is a nice workaround but I think this needs to be applied to other places as well.